pkg/termination/termination.go

```diff
@@ -37,38 +34,40 @@ import (
 )
 
 type Terminator struct {
-	EvictionQueue *EvictionQueue
-	KubeClient    client.Client
-	CloudProvider cloudprovider.CloudProvider
-	Clock         clock.Clock
+	clock         clock.Clock
+	kubeClient    client.Client
+	cloudProvider cloudprovider.CloudProvider
+	evictionQueue *EvictionQueue
 }
 
-type NodeDrainErr error
-
-func IsNodeDrainErr(err error) bool {
-	var nodeDrainErr NodeDrainErr
-	return errors.As(err, &nodeDrainErr)
+func NewTerminator(clk clock.Clock, kubeClient client.Client, cloudProvider cloudprovider.CloudProvider, eq *EvictionQueue) *Terminator {
```

Comment:
I'm not tracking why this doesn't live next to termination/controller.

Reply:
We are using this terminator and its flow in two places right now. We can eventually collapse it down when we remove this termination flow from the termination controller. But I would rather not have controller packages depend on each other and would opt to put something that is shared like this in a separate package.